<launch>
  <include
    file="$(find gazebo_ros)/launch/empty_world.launch" />
  <!-- Load joint controller configurations from YAML file to parameter server -->
  <rosparam file="$(find robot_do_an)/config/joint_names_robot_do_an.yaml" command="load"/>
  <!-- load the controllers -->
  <node name="controller_spawner" pkg="controller_manager" type="spawner" respawn="false"
    output="screen" ns="/robot_do_an" args="left_wheel_controller right_wheel_controller joint_state_controller"/>
  <!-- convert joint states to TF transforms for rviz, etc -->
  <node name="robot_state_publisher" pkg="robot_state_publisher" type="robot_state_publisher"
	  respawn="false" output="screen">
    <remap from="/joint_states" to="/robot_do_an/joint_states" />
  </node>
  <node
    name="tf_footprint_base"
    pkg="tf"
    type="static_transform_publisher"
    args="0 0 0 0 0 0 base_link base_footprint 40" />
  <node
    name="spawn_model"
    pkg="gazebo_ros"
    type="spawn_model"
    args="-file $(find robot_do_an)/urdf/robot_do_an.urdf -urdf -model robot_do_an"
    output="screen" />
  <node
    name="fake_joint_calibration"
    pkg="rostopic"
    type="rostopic"
    args="pub /calibrated std_msgs/Bool true" />
  
  <!-- <node
    name="keyboard_control"
    pkg="robot_do_an"
    type="keyboard_control.py"
    output="screen" />
  <node
    name="test_keyboard_control"
    pkg="robot_do_an"
    type="test_keyboard_control.py"
    output="screen" /> -->
</launch>